src/main/kotlin/com/cjbooms/fabrikt/generators/GeneratorUtils.kt

```diff
@@ -134,4 +142,70 @@ object GeneratorUtils {
         this.responses.filter { it.key != "default" }.values.filter(Response::hasContentMediaTypes)
 
     private fun Operation.filterParams(paramType: String): List<Parameter> = this.parameters.filter { it.`in` == paramType }
+
+    /**
+     * Returns a list of IncomingParameters, ordering logic should be
+     * encapsulated here to ensure the order of parameters align between
+     * services and controllers
+     */
+    fun Operation.toIncomingParameters(
+        basePackage: String,
+        pathParameters: List<Parameter>,
+        extraParameters: List<IncomingParameter>,
+    ): List<IncomingParameter> {
+
+        val bodies = requestBody.contentMediaTypes.values
+            .map {
+                BodyParameter(
+                    it.schema.safeName().toKotlinParameterName().ifEmpty { it.schema.toVarName() },
+                    requestBody.description,
+                    toModelType(basePackage, KotlinTypeInfo.from(it.schema)),
+                    it.schema
+                )
+            }
+
+        val parameters = mergeParameters(pathParameters, parameters)
+            .map {
+                RequestParameter(
+                    it.name,
+                    it.description,
+                    toModelType(basePackage, KotlinTypeInfo.from(it.schema), isNullable(it)),
+                    it
+                )
+            }
+            .sortedBy { it.type.isNullable }
+
+        val all = bodies + parameters + extraParameters
+
+        return if (all.map { it.name }.toSet().size != all.size) {
```

Comment:
Maybe extract this logic to a new function to avoid or reduce the necessary code comments.
eg, something like:
```
val allParameters = bodies + parameters + extraParameters
val distinctParameters = maybeRenameParametersToAvoidCoillisions(all)
return distinctParameters
```